gateleen-routing/src/main/java/org/swisspush/gateleen/routing/Router.java

```diff
@@ -221,6 +131,13 @@ public Router(Vertx vertx,
         this.httpClientFactory = httpClientFactory;
         this.doneHandlers = doneHandlers;
         this.routeMultiplier = routeMultiplier;
+        this.oAuthProvider = oAuthProvider;
+
+        if(oAuthProvider != null) {
+            this.oAuthStrategy = new OAuthStrategy(oAuthProvider);
+        }
+        this.basicAuthStrategy = new BasicAuthStrategy();
```

Comment:
This means that we can have two Strategies active, plan A and plan B. Shouldn't it be more one or the other?

Reply:
I the _RuleFactory_ class I added
```java
if (basicAuth != null && oAuthId != null) {
	throw new ValidationException("Either 'basicAuth' or 'oAuthId' can be given, not both");
}
```
on purpose. Normally we do already in the JSON schema enforce that only one of _basicAuth_ or _oAuthId_ can be provided. With this additional check I wanted to make it even more clear, that only one at a time is allowed. Also the unit test is more readable :-)